packages/webdriverio/src/utils/findStrategy.ts

```diff
@@ -232,12 +232,12 @@ export const findStrategy = function (selector: SelectorStrategy, isW3C?: boolea
     }
     case 'name': {
         if (isMobile || !isW3C) {
-            const match = stringSelector.match(/^\[name=("|')([a-zA-z0-9\-_.@=[\] ']+)("|')]$/)
+            const match = stringSelector.match(/^\[name=(("([^"]*)")|('([^']*)'))]$/)
             if (!match) {
                 throw new Error(`InvalidSelectorMatch. Strategy 'name' has failed to match '${stringSelector}'`)
             }
             using = 'name'
-            value = match[2]
+            value = match[3] || match[5]
```

Comment:
Just to be sure, I can't really tell how the match could have changed. Perhaps I am overlooking something but could you please explain this so we can be sure this is indeed correct?

The reason I ask is because it is still a single match right?

Reply:
There are two possibilities: matching with double quotes or matching with single quotes. Both have different capture groups (group 3 and group 5) in the regular expression. It would be possible to make it a single capture with a single regular expression as before, but then the regular expression would also match e.g. `[name="test']` which is an invalid expression. The other option is to make 2 different regular expressions, one for double quotes and one for single quotes. I thought this would be easiest: get the capture from the double quotes if it exists (capture 3), otherwise get the capture from the single quotes (capture 5).

Reply:
I see now, I feel like we should optimize the regex a bit.
```js
\[name=(?:"(.[^"]*)"|'(.[^']*)')]
```
It would be more performant and returns a match on the full regex and 1 group.

What do you think about this?

Reply:
Thanks, pushed an update